content-based-message-routing/README.md

```diff
@@ -0,0 +1,476 @@
+    ```ballerina
+    http:Client? hospitalEP = ();
+    ```
+- Use a [typed binding pattern with a mapping binding pattern to destructure](https://ballerina.io/learn/by-example/rest-binding-pattern-in-mapping-binding-pattern/) the payload and assign required components of the value to separate variables.
+    
+    ```ballerina
+    ReservationRequest {hospital_id, patient, ...reservationRequest} = payload;
+    ```
+    
+    Here,
+    - The `hospital_id` value is assigned to the `hospital_id` variable. 
+    - The `patient` value is assigned to the `patient` variable of type `Patient` (fields are exactly those expected by the `Patient` record). 
+    - The remaining components of the payload are assigned to the `reservationRequest` variable.
+
+- The `log` functions are used to [log](https://ballerina.io/learn/by-example/#log) information at `INFO`, `DEBUG` and `ERROR` log levels.
+
+- Use a [match statement](https://ballerina.io/learn/by-example/match-statement) to assign the relevant client to the `hospitalEP` variable based on the `hospital_id` value.
+    ```ballerina
+    match hospital_id {
+        "grandoaks" => {
+            log:printInfo("Routed to Grand Oak Community Hospital");
+            hospitalEP = grandOaksEP;
+        }
+        "clemency" => {
+            log:printInfo("Routed to Clemency Medical Center");
+            hospitalEP = clemencyEP;
+        }
+        "pinevalley" => {
+            log:printInfo("Routed to Pine Valley Community Hospital");
+            hospitalEP = pineValleyEP;
+        }
+    }
+    ```
+
+- After assigning the relevant client to the `hospitalEP` variable, check whether the variable is still null. If so, return a "NotFound" response.
+    ```ballerina
+    if hospitalEP is () {
+        log:printError(string `Routed to none. Hospital not found: ${hospital_id}`);
+        return <http:NotFound> {body: string `Hospital not found: ${hospital_id}`};
+    }
+    ```
+
+- The backend call is a `POST` request to the hospital service to reserve the appointment. The `category` value is used as path parameter. The response is assigned to the `resp` variable.
+    ```ballerina
+    ReservationResponse|http:ClientError resp = hospitalEP->/[category]/reserve.post({
+        patient, 
+        ...reservationRequest
+    });
+    ```
+
+- Use the `is` check to check whether the response is a `ReservationResponse` and return it as such if it is.
+    ```ballerina
+    if resp is ReservationResponse {
+        log:printDebug("Reservation request successful",
+                        name = patient.name,
+                        appointmentNumber = resp.appointmentNumber);
+        return resp;
+    }
+    ```
+- If the response is not a `ReservationResponse`, log the error and return a "NotFound" response if the response is a `http:ClientRequestError` or an "InternalServerError" response if the response is a `http:ServerError`.
```

Comment:
```suggestion
- If the response is not a `ReservationResponse`, log information at `ERROR` level and return a "NotFound" response if the response is a `http:ClientRequestError` or an "InternalServerError" response if the response is a `http:ServerError`.
```

I think we can improve the error stuff by rethinking the sentences. I did the same for service orchestration too.